src/shared/Microsoft.Git.CredentialManager.Tests/EnsureArgumentTests.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using Xunit;
+
+namespace Microsoft.Git.CredentialManager.Tests
+{
+    public class EnsureArgumentTests
+    {
+        [Theory]
+        [InlineData(5, 0, 10, true, true)]
+        [InlineData(0, 0, 10, true, true)]
+        [InlineData(10, 0, 10, true, true)]
+        [InlineData(0, -10, 0, true, true)]
+        [InlineData(-10, -10, 0, true, true)]
+        public void EnsureArgument_InRange_DoesNotThrow(int x, int lower, int upper, bool lowerInc, bool upperInc)
+        {
+            EnsureArgument.InRange(x, nameof(x), lower, upper, lowerInc, upperInc);
+        }
+
+        [Theory]
+        [InlineData(-3, 0, 10, true, true)]
+        [InlineData(13, 0, 10, true, true)]
+        [InlineData(10, 0, 10, true, false)]
+        [InlineData(0, 0, 10, false, true)]
+        [InlineData(-10, -10, 0, false, true)]
+        [InlineData(0, -10, 0, true, false)]
+        public void EnsureArgument_InRange_ThrowsException(int x, int lower, int upper, bool lowerInc, bool upperInc)
```

Comment:
Nit: should the name of this test be EnsureArgument_OutOfRange_ThrowsException() ??

Reply:
The naming pattern for the unit tests is `Type_Method_Input_Result`, where:

```
type   = EnsureArgument
method = InRange
input  = (null, it's all the different theory data)
result = throws an exception
```